src/components/AccordionItem.tsx

```diff
@@ -15,25 +15,14 @@ type Props = DivAttributes & {
     dangerouslySetExpanded?: boolean;
 };
 
-const defaultProps = {
-    className: 'accordion__item',
-};
-
-export default class AccordionItem extends React.Component<Props> {
-    static defaultProps: typeof defaultProps = defaultProps;
-
-    static displayName: DisplayName.AccordionItem = DisplayName.AccordionItem;
-
-    instanceUuid: UUID = nextUuid();
-
-    renderChildren = (itemContext: ItemContext): JSX.Element => {
-        const {
-            uuid,
-            className,
-            activeClassName,
-            dangerouslySetExpanded,
-            ...rest
-        } = this.props;
+const AccordionItem = ({
+    uuid = nextUuid(),
```

Comment:
Without a uuid this would set a default value of `nextUuid()` which would change on every rerender.

So we'd need a `useState` instead to preserve a generated id.

ie,

```typescript
const AccordionItem = ({
    uuid = nextUuid(),
 
Without a uuid this would set a default value of `nextUuid()` repeatedly on every render, which means the uuid would change on every rerender.

So instead we'd need to store the generated uuid somewhere such as a `useState` to preserve a generated id.

ie, something like...

```typescript
const AccordionItem = ({
    uuid: customUuid,
    dangerouslySetExpanded,
    className = 'accordion__item',
    activeClassName,
    ...rest
}: Props): JSX.Element => {
  const [generatedUuid] = useState(nextUuid());
  const uuid = customUuid || generatedUuid;
  // ...
```

Reply:
btw this exact bug occurred a few months ago too! https://github.com/springload/react-accessible-accordion/pull/285/files#diff-431e4583f47c237f09bde4a4d9caedc7808a26d1dc40567edfdfd0b91a25c947L19

Reply:
Ahh I see! Cool I'll fix that up